CONTRIBUTING.md

```diff
@@ -1,6 +1,21 @@
 All informations on contributing can be found on the [corresponding wiki page.](https://github.com/iScsc/wiresmash/wiki/Contributing) Read at least the general policies section, and specific sections. 
 
 Quick TLDR:
+- **FORMAT CODE BEFORE DOING A PULL REQUEST**, see [corresponding section](#formatting-code).
 - To contribute, do fork-branch-PR.
 - To report bugs, make sure not to create duplicates and create an issue with as many information as possible. Name it `bug-xxx`.
 - For feature requests, make an issue named `request-xxx`
+
+## Formatting code<a name="formatting"></a>
+
+It is advised to format code either as you go, but *atleast once* before doing a PR. PR wil provide a fail if the code is not formatted. **Only format files on which you worked.** We use `clang-format`, version `16+` for formatting, and the file is already in the repo (`.clang-format` in root).
+
+A pre-commit hook is provided in `hooks/format-hook`. It is strongly advised that you move the code inside of it to the `.git/hooks/pre-commit` file. It will warn you when you try to commit unformatted code.
+
+To run `clang-format` on a single file, simply run from the root of the repo the following command:
+```bash
+clang-format -i path/to/file.extension
```

Comment:
i'm not sure i understand the procedure to follow :thinking: 

- how to format the whole workspace?
- `git clang-format` only formats staged files, right?

maybe adding a `Makefile` with a few simple rules could be great to avoid confusion
- `fmt` and `fmt-check` rules
- mention the `make fmt` in the error output of the githook

Reply:
You cannot format the whole workspace, clang-format only allows to format one file at a time, and I will trust the makers of clang on the fact that there are good reason as to why.

You have several ways to setup formatting, I just added the main ones:
- How to run the `clang-format` command, so ppl can just "have it" instead of looking a the doc
- An easier way to run it with git automatically picking staged files.
- You can also for example turn on auto format ok save with the vscode `C/C++` extension

I personally don't want to have too much make rules to keep the development workflow cross-platform, plus in this case I don't really know how you would feel confused, the commands are fairly simple. But if we end up having explanation for everything using "normal" workflow and then "Or use this make rule" it would be weird, but just my opinion

Reply:
i think we need some sort of script, either a `.sh` script of `make` rules to
1. check the format
2. run the formatter and modify the bad files

we do not want to rely on the devs (1) understanding how this all works and (2) applying the formatting correctly :thinking: 

i do think it's far better to have `fmt` and `fmt-check` to do all that for them :relieved:

Reply:
The pre-commit hook does 1 and `git clang-format` does 2 so I still believe it only add a possible linux-only workflow

Reply:
okey that's better now :+1: